lldb/source/Plugins/SymbolFile/DWARF/DWARFASTParserSwiftDescriptorFinder.cpp

```diff
@@ -249,6 +253,31 @@ DWARFASTParserSwift::getBuiltinTypeDescriptor(
       type.GetMangledTypeName());
 }
 
+NodePointer DWARFASTParserSwift::FindSuperclass(DWARFDIE &die) {
+  const auto inheritance_die_it =
+      llvm::find_if(die.children(), [&](const DWARFDIE &child_die) {
+        return child_die.Tag() == llvm::dwarf::DW_TAG_inheritance;
+      });
+
+  if (inheritance_die_it == die.children().end())
+    return nullptr;
+
+  auto inheritance_die = *inheritance_die_it;
+
+  const auto superclass_type_die =
+      inheritance_die.GetAttributeValueAsReferenceDIE(llvm::dwarf::DW_AT_type);
```

Comment:
I think we could probably get more reuse out of this function if we split it into a FindSuperClassDIE function and a GetCanonicalDemangleTree(DIE) function